### PR TITLE
Fix issues in FontCharsClass::Create_GDI_Font

### PR DESCRIPTION
**Pull request by xezon**

### CODE REVIEW

File: src/w3d/renderer/render2dsentence.cpp [hunk 1]
```diff
@@ -246,20 +246,22 @@ void FontCharsClass::Create_GDI_Font(const char *font_name)
...
-    bitmap_info.biClrUsed = 0;
-    bitmap_info.biClrImportant = 0;
-    m_gdiBitmap = CreateDIBSection(
-        screen_dc, (const BITMAPINFO *)&bitmap_info, DIB_RGB_COLORS, (void **)&m_gdiBitmapBits, nullptr, 0L);
+    // #BUGFIX Do not pass a BITMAPINFOHEADER* for BITMAPINFO* into CreateDIBSection function.
+
+    BITMAPINFO bitmap_info = { 0 };
+    bitmap_info.bmiHeader.biSize = sizeof(BITMAPINFOHEADER);
+    bitmap_info.bmiHeader.biWidth = m_pointSize * 2;
+    bitmap_info.bmiHeader.biHeight = m_pointSize * -2;
```
Comment (xezon) [inline src/w3d/renderer/render2dsentence.cpp]: Was `~2` before. That would be -3. But WorldBuilder shows -2 here.

File: src/w3d/renderer/render2dsentence.cpp [hunk 2]
```diff
@@ -226,10 +226,10 @@ void FontCharsClass::Create_GDI_Font(const char *font_name)
     }
 
     if (is_generals) {
-        font_width = ((double)-font_height / 2.5);
+        font_width = ((float)-font_height * 0.4f);
     }
 
-    m_widthReduction = std::clamp(font_height / ~7, 0, 4);
+    m_widthReduction = std::clamp(font_height / -8, 0, 4);
```
Comment (xezon) [inline src/w3d/renderer/render2dsentence.cpp]: ~7 is -8, and WorldBuilder shows -8 here as well.

File: src/w3d/renderer/render2dsentence.cpp (hunk 1 above)
Comment (tomsons26) [reply src/w3d/renderer/render2dsentence.cpp]: -(m_pointSize * 2); actually

File: src/w3d/renderer/render2dsentence.cpp [hunk 3]
```diff
@@ -226,10 +226,10 @@ void FontCharsClass::Create_GDI_Font(const char *font_name)
     }
 
     if (is_generals) {
-        font_width = ((double)-font_height / 2.5);
+        font_width = ((float)-font_height * 0.4f);
```
Comment (tomsons26) [inline src/w3d/renderer/render2dsentence.cpp]: / 2.5 is the same thing

File: src/w3d/renderer/render2dsentence.cpp (hunk 1 above)
Comment (xezon) [reply src/w3d/renderer/render2dsentence.cpp]: It's same result but yes I agree that reads better. I will change.

File: src/w3d/renderer/render2dsentence.cpp (hunk 3 above)
Comment (xezon) [reply src/w3d/renderer/render2dsentence.cpp]: I know. Do you want to keep the `2.5` ? World Builder shows * 0.4f

Comment (tomsons26) [reply src/w3d/renderer/render2dsentence.cpp]: yes, i can just divide with/by 1 to find out what it should be so its not really a issue for comparing to pseudo
figuring out the real value is preferred for floats/doubles
2.5 is what the actual code would have had, it has a meaning, the multiplications compilers optimize divisions to can end up being cryptic

File: src/w3d/renderer/render2dsentence.cpp [hunk 4]
```diff
@@ -226,10 +226,10 @@ void FontCharsClass::Create_GDI_Font(const char *font_name)
...
     }
 
-    m_widthReduction = std::clamp(font_height / ~7, 0, 4);
+    m_widthReduction = std::clamp(-(font_height / 8), 0, 4);
```
Comment (tomsons26) [inline src/w3d/renderer/render2dsentence.cpp:232]: where did - come from?

**Review by tomsons26 — APPROVED** [review]

**Pull request merged**